python/src/prima/examples/cobyla/cobyla_example.py

```diff
@@ -0,0 +1,322 @@
+import numpy as np
+from prima.common.evaluate import evaluate
+from prima.common.consts import EPS, RHOBEG_DEFAULT, RHOEND_DEFAULT, \
+    CTOL_DEFAULT, CWEIGHT_DEFAULT, FTARGET_DEFAULT, IPRINT_DEFAULT, \
+    MAXFUN_DIM_DEFAULT
+from prima.cobyla.cobylb import cobylb
+
+debugging = True
+
+def test_chebyquad():
+    f, _ = calcfc_chebyquad([1, 2])
+    assert np.isclose(f, 91+1/9, atol=1e-6)
+
+
+def calcfc_chebyquad(x):
+    x = np.array(x)
+    n = len(x)  # or shape?
+    y = np.zeros((n + 1, n + 1))
+    y[0:n, 0] = 1
+    y[0:n, 1] = 2*x - 1
+    for i in range(1, n):
+        y[0:n, i+1] = 2*y[0:n, 1] * y[0:n, i] - y[0:n, i - 1]
+
+    f = 0
+    for i in range(n+1):
+        tmp = sum(y[0:n, i]) / n
+        if i % 2 == 0:
+            tmp += 1/((i)**2 - 1)
+        f += tmp**2
+    constr = 0
+    return f, constr
+
+def calcfc_hexagon(x):
+    # Test problem 10 in Powell's original algorithm
+    
+    assert len(x) == 9
+    
+    f = -0.5 * (x[0] * x[3] - x[1] * x[2] + x[2] * x[8] - x[4] * x[8] + x[4] * x[7] - x[5] * x[6])
+    constr = np.zeros(14)
+    constr[0] = 1 - x[2]**2 - x[3]**2
+    constr[1] = 1 - x[8]**2
+    constr[2] = 1 - x[4]**2 - x[5]**2
+    constr[3] = 1 - x[1-1]**2 - (x[1] - x[8])**2
+    constr[4] = 1 - (x[1-1] - x[4])**2 - (x[1] - x[5])**2
+    constr[5] = 1 - (x[1-1] - x[6])**2 - (x[1] - x[7])**2
+    constr[6] = 1 - (x[2] - x[4])**2 - (x[3] - x[5])**2
+    constr[7] = 1 - (x[2] - x[6])**2 - (x[3] - x[7])**2
+    constr[8] = 1 - x[6]**2 - (x[7] - x[8])**2
+    constr[9] = x[1-1] * x[3] - x[1] * x[2]
+    constr[10] = x[2] * x[8]
+    constr[11] = -x[4] * x[8]
+    constr[12] = x[4] * x[7] - x[5] * x[6]
+    constr[13] = x[8]
+
+    return f, constr
+
+def cobyla(calcfc, m, x, f, cstrv=0, constr=0, f0=None, constr0=None, nf=None, rhobeg=None,
```

Comment:
imho this should be part of the module, not the example, ie

```
import prima
x,f = prima.cobyla(...)
```

Reply:
Yes it will be, this was just me "slinging code" before ultimately re-arranging it. This PR is more for transparency and awareness of progress than for detailed code comments. Once the work is more mature I will close this PR and open a second one for more detailed code comments along the lines of this one.